### PR TITLE
Failed plugin state added. When plugin failed to start previous state was kept

### PR DESCRIPTION
**Pull request by radeklos**
There is no easy way how to distinguish FAILED and STOPPED/RESOLVED plugins.

### CODE REVIEW

File: pf4j/src/main/java/org/pf4j/AbstractPluginManager.java [hunk 1]
```diff
@@ -334,6 +334,7 @@ public void startPlugins() {
 
                     firePluginStateEvent(new PluginStateEvent(this, pluginWrapper, pluginState));
                 } catch (Exception e) {
+                    pluginWrapper.setPluginState(PluginState.FAILED);
```
Comment (decebals) [inline pf4j/src/main/java/org/pf4j/AbstractPluginManager.java]: I think that we need also to fire the plugin state event with:
```
firePluginStateEvent(new PluginStateEvent(this, pluginWrapper, pluginState));
```
(probably to move the call of `firePluginStateEvent` outside of the try/catch block).

Also I feel that we should keep/save the problem/exception for late use, but I don't know where:
- a property `failedException` in `PluginWrapper`
- a property (Map, ...) in `AbstractPluginManager`, something similar with `resolvedPlugins`, `startedPlugins`, ... 
- other variant

Comment (radeklos) [reply pf4j/src/main/java/org/pf4j/AbstractPluginManager.java]: I created the property. Not sure whether I need to clean it up somewhere else.

**Comment by decebals** [comment]
Thanks!

File: pf4j/src/main/java/org/pf4j/AbstractPluginManager.java (hunk 1 above)
Comment (decebals) [reply pf4j/src/main/java/org/pf4j/AbstractPluginManager.java]: > I created the property. Not sure whether I need to clean it up somewhere else.

Probably the best place is in [PluginState#setPluginState](https://github.com/pf4j/pf4j/blob/master/pf4j/src/main/java/org/pf4j/PluginWrapper.java#L137) if `PluginState != FAILED` but for the moment I think that it's OK.

**Pull request merged**